neural_tangents/stax.py

```diff
@@ -1083,9 +1085,9 @@ def apply_fn(params, inputs, **kwargs):
   else:
     def rescaler(dims, strides, padding):
       del dims, strides, padding  # Unused.
-      return lambda outputs, inputs, spec: outputs / np.prod(window_shape)
+      return lambda outputs, inputs, spec: outputs * np.prod(window_shape)
```

Comment:
Why switch from division to multiplication here?

Reply:
For TensorFlow `pool`, it only has average pooling, so when doing sum pooling, it needs to be multiplied by the window size, which is opposite from JAX `reduce_window`.

Reply:
1) Great point! I believe in that case it's best to have the multiplication inside the `lax.reduce_window`, for the purpose of matching the original `lax.reduce_window` behavior (I believe it's called in other places too, that way you don't need to adjust the computation in such places)

2) I also think this branch is when `pool_type == Pooling.AVG`, so in that case, you would want to have no rescaling here, correct (instead, it currently behaves as a sum pool, not average).

Reply:
Thanks for bringing this up, Roman! Do you have an idea on how to embed sum pooling inside `reduce_window`? The only parameter related to pooling type that `reduce_window` receives is the third parameter, which is a callable (that is, either `np.add` or `np.max`). For both sum pooling and average pooling, `np.add` is the representation for them, and I think this is also the same case for JAX? That is the reason that in `ostax`, an extra rescaler function is used to normalize the output when it is average pooling (JAX `reduce_window` does not automatically support average pooling, while TF `reduce_window` does not automatically support sum pooling).

Please correct me if I am wrong.

Reply:
> I also think this branch is when pool_type == Pooling.AVG, so in that case, you would want to have no rescaling here, 
> correct (instead, it currently behaves as a sum pool, not average).

Yes, I think you are right!

Reply:
Yes your understanding is correct, and we just want your version of `lax.reduce_window` to return sum pooling result. So you can just have `* np.prod(window_shape)` inside your `lax.reduce_window`, and all other places, like this one, wouldn't need to be changed.

Reply:
I.e. the decision to normalize or not is made in other places around the code, by just using a rescaler; `lax.reduce_window` just needs to return the sum.

Reply:
Oh, I see what you mean, Roman! Sorry I misunderstood you. @romanngg